src/mlpack/methods/amf/termination_policies/simple_residue_termination.hpp

```diff
@@ -53,7 +53,7 @@ class SimpleResidueTermination
   {
     // Initialize the things we keep track of.
     residue = DBL_MAX;
-    iteration = 1;
+    iteration = 0;
```

Comment:
I changed the initial `iteration` to 0, so the iteration number printed by `Log::Info` below will start from `1`. If `iteration` is initially set to 1, the starting iteration number would be `2`.

Reply:
Ah, ok, I see.  The trick we do for other optimizers does not work in this case because `iteration++` must come before the return statement.  Another option might be to remove the `iteration++` from line 81 and embed it into the conditional:

```
return (residue < minResidue || iteration++ != maxIterations);
```

but I am not picky, I think whatever you choose is ok.

Reply:
Do you mean `return (residue < minResidue || iteration++ >= maxIterations);`? Because like I mentioned in a previous comment, `iteration != maxIterations` doesn't work properly when `maxIterations != 0`.
And I think keeping `iteration++` in line 81 is more clear and readable.

Reply:
Ack, sorry, the condition would actually be `iteration++ == maxIterations`.  Anyway, I can agree that `iteration++` in line 81 is more clear, so we can just keep it like that.

Reply:
What if `iteration` overflows and becomes 0?
But it seems very unlikely...

Reply:
Yep, we are making the assumption in the other code that the user isn't going to be patient enough to wait for 2^64 iterations (or 2^32 depending on architecture).

Reply:
modified, should be ready to merge now.